### DISCUSSION TITLE
restrictive color profile when import MNG file

### DISCUSSION THREAD

**Discussion by Vandalhart** [dlemstra/Magick.NET#1714] [Help]
I'm back again.

So I'm attempting to import an existing MNG, convert to a PNG then use those PNGs as Texture 2Ds and grabbing/moving pixels around to make a similar animation.

Now, I have the second step working, but I've noticed that it appears that when I import my .MNG it restricts the color profile of import frames to what I'm assuming is 2 colors.

`            using MagickImageCollection collectionMNG = new MagickImageCollection(Application.dataPath + "importFile.MNG");
            {

                foreach (var item in collectionMNG)
                {
                    var fileName = $"importFile_{frame}.png";
                    filePath = System.IO.Path.Combine(Application.dataPath, "../Renders/");
                    filePath = System.IO.Path.Combine(filePath, fileName);

                    item.Write(filePath, MagickFormat.Png);
                    frame++;
                }`

So this WILL export my frames as PNGs, but I'm not sure what I can do about the files being two colors. 
This is how they should look:
![image](https://github.com/user-attachments/assets/febd0a52-4af0-47c9-8e7c-07341dc11a19)

This is how they look:
![image](https://github.com/user-attachments/assets/bec775fc-423f-4118-b01e-bdb167825a94)

Those are the same 4 frames in both preview images.

In some of my failed attempts I swear I saw something reference to a 2bit file maybe? I had tried to set the Magick.Format to a few different types to no success.

Any assistance would be appreciated.

**Comment by Vandalhart** [comment]
Strange, I swapped to a different file and it didn't do it when I attempted to change it out. I will keep an eye on it and update if I see the problem happen again.

**Comment by dlemstra** [comment]
Can you still reproduce this with your initial file and is it possible to share it?

**Comment by Vandalhart** [comment]
It hasn't happened again. When I went back through my code, I think I had updated a collection name but missed some lines that referenced it. Since correcting that it didn't seem to happen again.

**Comment by Vandalhart** [comment]
On a unrelated note to this issue, which has never cropped up again so I think it was most likely user error, is there a way to remove the alpha from my MNG files when I save? At the moment it's looking like I'd have to open them individually and set it Opacity/Matte color but I'm sure there's something within the collection settings that'd let me do it.

I've been scouring through the documentation but I'm just not seeing what I'm looking for.

**Reply by dlemstra** [reply]
If you need to remove the alpha channel you can do that with the `.Alpha(` method. And if you want to give the alpha channel a color you can use `.ColorAlpha(`.

**Reply by Vandalhart** [reply]
Thank you.

Is there any else I have to include at the top of my .cs file?
I have:
```
using System.IO;
using UnityEngine;
using ImageMagick;
using System;
```

at the top of my file and when I attempt to use collectionName.Alpha(); it doesn't pick up the .Alpha() method at all.

**Reply by dlemstra** [reply]
You need to do that for each individual file in the collection.

**Reply by Vandalhart** [reply]
wonderful, that was going to be my next question. Thank you very much.
